tests/test_show_diff_format.py

```diff
@@ -0,0 +1,57 @@
+import json
+
+
+def test_show_diff_format_none_json(cli_runner, data_archive):
+    # Check that the json output is a commit metadata only
+    with data_archive("points.tgz"):
+        r = cli_runner.invoke(["show", "--diff-format=none", "-o", "json"])
+        output = json.loads(r.stdout)
+
+        # Assert that the output is a dict with a key for metadata only
+        assert output.keys() == {"kart.show/v1"}
+
+
+def test_show_diff_format_no_data_changes_json(cli_runner, data_archive):
+    # Check that the json output is a commit metadata (if any) and a boolean for data_changes (feature/tile changes)
+    with data_archive("points.tgz"):
+        r = cli_runner.invoke(["show", "--diff-format=no-data-changes", "-o", "json"])
+        output = json.loads(r.stdout)
+        assert output["kart.diff/v1+hexwkb"] == {
+            "nz_pa_points_topo_150k": {"data_changes": True}
+        }
+
```

Comment:
I think we also need a test of a meta-only commit (ie `data_changes` will be False).

Look at `tests/data/meta-updates.tgz` - the latest commit in that repo has only meta changes